### PR TITLE
Define `etc/profile.d/mamba.sh` and install it

### PR DESCRIPTION
**Pull request by jjerphan**
Continues #3410.

### CODE REVIEW

**Comment by jjerphan** [comment]
This works and passes locally for me. 🤔

File: dev/environment-dev.yml [hunk 1]
```diff
@@ -39,7 +39,6 @@ dependencies:
   - securesystemslib
   # libmambapy build dependencies
   - scikit-build
-  - build
```
Comment (Hind-M) [inline dev/environment-dev.yml:42]: Why add this?

**Comment by Hind-M** [comment]
Why did you remove `mamba_package/etc/profile.d/mamba.sh` from `.gitignore`?

File: dev/environment-dev.yml (hunk 1 above)
Comment (jjerphan) [reply dev/environment-dev.yml:42]: [The `build` package on `conda-forge`](https://prefix.dev/channels/conda-forge/packages/build) has been yanked.

**Comment by jjerphan** [comment]
> Why did you remove `mamba_package/etc/profile.d/mamba.sh` from .gitignore?

`etc/profile.d/mamba.sh` is now generated in the build folder and not in the source. Since `build` is `.gitignored`, we do not need `mamba_package/etc/profile.d/mamba.sh` to be as well.

File: dev/environment-dev.yml (hunk 1 above)
Comment (Hind-M) [reply dev/environment-dev.yml:42]: Isn't it superseded by something else?

Comment (jjerphan) [reply dev/environment-dev.yml:42]: It is superseded by `python-build` (see https://github.com/conda-forge/conda-forge.github.io/issues/2269), but `python-build` itself is not required it seems.

**Review by Hind-M — APPROVED** [review]

**Pull request merged**